Bigtable/tests/System/UniverseDomainTest.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * Copyright 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Bigtable\Tests\System;
+
+use Google\ApiCore\ApiException;
+use Google\Cloud\Bigtable\Admin\V2\Client\BigtableInstanceAdminClient;
+use Google\Cloud\Bigtable\Admin\V2\Client\BigtableTableAdminClient;
+use Google\Cloud\Bigtable\Admin\V2\Cluster;
+use Google\Cloud\Bigtable\Admin\V2\ColumnFamily;
+use Google\Cloud\Bigtable\Admin\V2\CreateInstanceRequest;
+use Google\Cloud\Bigtable\Admin\V2\CreateTableRequest;
+use Google\Cloud\Bigtable\Admin\V2\DeleteInstanceRequest;
+use Google\Cloud\Bigtable\Admin\V2\DeleteTableRequest;
+use Google\Cloud\Bigtable\Admin\V2\GetInstanceRequest;
+use Google\Cloud\Bigtable\Admin\V2\GetTableRequest;
+use Google\Cloud\Bigtable\Admin\V2\Instance;
+use Google\Cloud\Bigtable\Admin\V2\Table;
+use Google\Cloud\Bigtable\BigtableClient;
+use Google\Cloud\Bigtable\Mutations;
+use Google\Cloud\Bigtable\V2\RowFilter;
+use Google\Cloud\Core\Testing\System\SystemTestCase;
+
+class UniverseDomainTest extends SystemTestCase
+{
+    private static $instanceAdminClient;
+    private static $tableAdminClient;
+    private static $bigtableClient;
+    private static $tableId;
+    private static $instanceId;
+    private static $clusterId;
+    private static $projectId;
+    private static $locationId;
+
+    /**
+     * @beforeClass
+     */
+    public static function setUpTestFixtures(): void
+    {
+        if (!$keyFilePath = getenv('GOOGLE_CLOUD_PHP_TESTS_UNIVERSE_DOMAIN_KEY_PATH')) {
+            self::markTestSkipped('Set GOOGLE_CLOUD_PHP_TESTS_UNIVERSE_DOMAIN_KEY_PATH to run system tests');
+        }
+
+        // This can be found for your universe by running "gcloud compute zones list"
+        if (!$locationId = getenv('GOOGLE_CLOUD_PHP_TESTS_UNIVERSE_DOMAIN_LOCATION')) {
+            self::markTestSkipped('Set GOOGLE_CLOUD_PHP_TESTS_UNIVERSE_DOMAIN_LOCATION to run system tests');
+        }
+
+        $credentials = json_decode(file_get_contents($keyFilePath), true);
```

Comment:
nit:
json_decode can return null. I wouldn't worry too much about it in here though.

Reply:
how about this:

```suggestion
        if (!$credentials = json_decode(file_get_contents($keyFilePath), true)) {
            throw new \Exception('unable to decode key file');
        }
```